src/main/cpp/chassis/states/DriveToCoralStation.h

```diff
@@ -16,39 +16,26 @@
 #pragma once
 
 // C++ Includes
-#include <vector>
+#include <string>
 
 // FRC Includes
-#include <frc/geometry/Rotation3d.h>
-#include <frc/geometry/Rotation2d.h>
-#include <frc/geometry/Pose2d.h>
 
 // Team302 Includes
-#include "chassis/states/RobotDrive.h"
-#include "vision/DragonVision.h"
+#include "chassis/states/DriveToFieldElement.h"
 #include "fielddata/DragonTargetFinder.h"
-#include "pathplanner/lib/trajectory/PathPlannerTrajectory.h"
-#include "chassis/states/TrajectoryDrivePathPlanner.h"
-#include "utils/FMSData.h"
-#include "chassis/SwerveChassis.h"
 
-class DriveToCoralStation : public TrajectoryDrivePathPlanner
+class RobotDrive;
+class TrajectoryDrivePathPlanner;
+
+class DriveToCoralStation : public DriveToFieldElement
 {
 public:
     DriveToCoralStation(RobotDrive *robotDrive, TrajectoryDrivePathPlanner *trajectoryDrivePathPlanner);
     std::string GetDriveStateName() const override;
 
-    pathplanner::PathPlannerTrajectory CreateTrajectory(std::optional<std::tuple<DragonTargetFinderData, frc::Pose2d>> info);
-
-    void Init(ChassisMovement &chassisMovement) override;
-    void InitFromTrajectory(ChassisMovement &chassisMovement, pathplanner::PathPlannerTrajectory trajectory) override;
-    pathplanner::PathPlannerTrajectory GetTrajectory() const { return m_trajectory; }
-    std::array<frc::SwerveModuleState, 4> UpdateSwerveModuleStates(ChassisMovement &chassisMovement) override;
-
-private:
-    pathplanner::PathPlannerTrajectory CreateDriveToCoralStationTrajectory(frc::Pose2d currentPose, frc::Pose2d csaPose);
-
-    pathplanner::PathPlannerTrajectory m_trajectory;
-    DragonTargetFinderData m_currentType = DragonTargetFinderData::NOT_FOUND;
-    frc::Pose2d m_endPose;
+protected:
+    DragonTargetFinderTarget GetDriveToTarget() const override;
+    ChassisOptionEnums::DriveStateType GetDriveStateType() const override;
+    ChassisOptionEnums::HeadingOption GetHeadingOption() const override;
+    units::angle::degree_t GetModifiedHeadingValue(units::angle::degree_t calculatedHeading) { return (calculatedHeading - 180_deg); }
```

Comment:
This needs to be in the DriveTo Reef Branches, Coral station we don't apply the 180 offset

Reply:
Modified formula for Coral and default drive to field element